reconcile/openshift_resources_base.py

```diff
@@ -817,3 +858,79 @@ def run(
         sys.exit(1)
 
     return ri
+
+
+def early_exit_desired_state(
+    providers: list[str], resource_schema_filter: Optional[str] = None
+) -> dict[str, Any]:
+    settings = queries.get_secret_reader_settings()
+    namespaces, _ = get_namespaces(
+        providers, resource_schema_filter=resource_schema_filter
+    )
+    fetch_specs = [
+        (r, ns_info) for ns_info in namespaces for r in ns_info["openshiftResources"]
+    ]
+
+    with early_exit_monkey_patch():
+        resources = threaded.run(
+            get_resource,
+            fetch_specs,
+            thread_pool_size=10,
+            settings=settings,
+        )
+
+    return {
+        "namespaces": namespaces,
+        "resources": resources,
+    }
+
+
+def get_resource(spec, settings):
+    resource = spec[0]
+    ns_info = spec[1]
+    if resource.get("enable_query_support"):
+        c = fetch_openshift_resource(
+            resource, ns_info, skip_validation=True, settings=settings
+        ).body
+    else:
+        c = resource["resource"].get("content")
+    if not c:
+        print(resource)
+    del resource["resource"]
+    return c
+
+
+@contextmanager
+def early_exit_monkey_patch():
+    """Avoid looking outside of app-interface on early-exit pr-check."""
+    orig_lookup_secret = lookup_secret
+    orig_lookup_github_file_content = lookup_github_file_content
+    orig_url_makes_sense = url_makes_sense
+    orig_check_alertmanager_config = check_alertmanager_config
+
+    try:
+        yield early_exit_monkey_patch_assign(
+            lambda path, key, version=None, tvars=None, settings=None: f"vault({path}, {key}, {version})",
+            lambda repo, path, ref, tvars=None, settings=None: f"github({repo}, {path}, {ref})",
+            lambda url: False,
+            lambda data, path, alertmanager_config_key, decode_base64=False: True,
+        )
+    finally:
+        early_exit_monkey_patch_assign(
+            orig_lookup_secret,
+            orig_lookup_github_file_content,
+            orig_url_makes_sense,
+            orig_check_alertmanager_config,
+        )
+
+
+def early_exit_monkey_patch_assign(
+    lookup_secret,
+    lookup_github_file_content,
+    url_makes_sense,
+    check_alertmanager_config,
+):
+    sys.modules[__name__].lookup_secret = lookup_secret
+    sys.modules[__name__].lookup_github_file_content = lookup_github_file_content
+    sys.modules[__name__].url_makes_sense = url_makes_sense
+    sys.modules[__name__].check_alertmanager_config = check_alertmanager_config
```

Comment:
why this change?

Reply:
the original code looked like
```
orb.lookup_secret = lookup_secret
```
since we are now in the `orb` module itself, it needed to be rewritten. so i tried
```
lookup_secret = patched_lookup_secret
```
but then vscode syntax highlighting was broken. it did not show, that `lookup_secret` would be the module function that is currently being overwritten. this might be just me but i rely heavily on syntax and color highlighting to get a feeling what is going on.

so i decided to be very explicit and that is how i landed with 

```
sys.modules[__name__].lookup_secret = lookup_secret
```

i'm open for suggestions

Reply:
Fair enough :)